clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/vmimage/view/AzureVMImageLookupController.groovy

```diff
@@ -90,50 +125,67 @@ class AzureVMImageLookupController {
   List<AzureNamedImage> list(LookupOptions lookupOptions) {
     def result = new ArrayList<AzureNamedImage>()
 
-    // return a list of virtual machine images as read from the config.yml file
-    accountCredentialsProvider.getAll().each { creds ->
-      if(creds instanceof AzureNamedAccountCredentials) {
-        creds.vmImages.each { vmImage ->
-          def imageName = vmImage.offer + "-" + vmImage.sku + " (Recommended)"
-          if (lookupOptions.q == null || imageName.toLowerCase().startsWith(lookupOptions.q.toLowerCase())) {
-            result += new AzureNamedImage(
-              imageName: imageName,
-              publisher: vmImage.publisher,
-              offer: vmImage.offer,
-              sku: vmImage.sku,
-              version: vmImage.version
-            )
-            if (result.size() >= MAX_SEARCH_RESULTS)
-              return result
+    // retrieve the list of custom vm images from the SCS specified in the config.yml file and stored in the cache
+    result.addAll(
+      getAllMatchingKeyPattern(
+        lookupOptions.q,
+        Keys.Namespace.AZURE_CUSTOMVMIMAGES.ns,
+        Keys.getCustomVMImageKey(azureCloudProvider,
+          lookupOptions.account ?: '*',
+          lookupOptions.region ?: '*',
+          "*"),
+        true))
+
+
+    if (!lookupOptions.customOnly) {
+      // return a list of virtual machine images as read from the config.yml file
+      accountCredentialsProvider.getAll().each { creds ->
+        if (creds instanceof AzureNamedAccountCredentials && (!lookupOptions.account || lookupOptions.account.isEmpty() || lookupOptions.account == creds.accountName)) {
+          creds.vmImages.each { vmImage ->
+            def imageName = vmImage.offer + "-" + vmImage.sku + "(Recommended)"
+            if (lookupOptions.q == null || imageName.toLowerCase().contains(lookupOptions.q.toLowerCase())) {
```

Comment:
This should be 
`if (!lookupOptions.q || ...)`
otherwise, `lookupOptions.q == ""` will always be contained in `imageName`, which might not be expected behavior.